e2e/suite_test.go

```diff
@@ -0,0 +1,181 @@
+package e2e_test
```

Comment:
future readability suggestion: would moving the specific test cases (and the suite run) to a subfolder in e2e/ make it easier to read? And having one test per file? since we plan on adding more test cases

Reply:
I think we can refactor most of the code still yes :D